draft-ietf-mls-protocol.md

```diff
@@ -4352,71 +4352,75 @@ current state of the group.  In practice, however, there is a risk
 that two members will generate Commit messages simultaneously,
 based on the same state.
 
-When this happens, there is a need for the members of the group to
-deconflict the simultaneous Commit messages.  There are two
-general approaches:
-
-* Have the Delivery Service enforce a total order
-* Have a signal in the message that clients can use to break ties
-
-As long as Commit messages cannot be merged, there is a risk of
-starvation.  In a sufficiently busy group, a given member may never
-be able to send a Commit message, because he always loses to other
-members.  The degree to which this is a practical problem will depend
+The extent to which this is a problem, and the appropriate solution, depends on
+the design of the Delivery Service. Per the CAP theorem, there are two general
+classes of distributed system:
+
+* Consistent and Partition-tolerant, or CP, systems can provide a globally
+  consistent view of data but may stop working if there are network issues.
+* Available and Partition-tolerant, or AP, systems continue working despite
+  network issues but may return different views of data to different users.
+
+Strategies for sequencing messages in CP and AP systems are described in the
+next two sections. Implementations MUST only update their cryptographic state
+when they have a reasonable degree of certainty that their Commit was accepted.
+The generation of Commit messages MUST NOT modify a client's state, since the
+client doesn't know at that time whether the changes implied by the Commit
+message will conflict with another Commit or not. Similarly, a client MUST NOT
+send Welcome messages corresponding to a Commit until it's clear that the Commit
+has been accepted.
+
+Regardless of how messages are kept in sequence, there is a risk that
+in a sufficiently busy group, a given member may never
+be able to send a Commit message because they always lose to other
+members. The degree to which this is a practical problem will depend
 on the dynamics of the application.
 
-It might be possible, because of the non-contributivity of intermediate
-nodes, that Commit messages could be applied one after the other
-without the Delivery Service having to reject any Commit message,
-which would make MLS more resilient regarding the concurrency of
-Commit messages.
-The Messaging system can decide to choose the order for applying
-the state changes. Note that there are certain cases (if no total
-ordering is applied by the Delivery Service) where the ordering is
-important for security, ie. all updates must be executed before
-removes.
-
-Regardless of how messages are kept in sequence, implementations
-MUST only update their cryptographic state when valid Commit
-messages are received.
-Generation of Commit messages MUST NOT modify a client's state, since the
-endpoint doesn't know at that time whether the changes implied by
-the Commit message will succeed or not.
-
-## Server-Enforced Ordering
+## CP Ordering
```

Comment:
No thank you. Please do not a use a relatively rare acronym when the reading audience will understand "Server-Enforced Ordering" just fine.